pkg/certgen/certgen.go

```diff
@@ -55,7 +57,11 @@ func VerifyServiceCertAndKey(fileMap map[string][]byte, fileNamePrefix string, c
 		return errors.New("unparseable certificate in file map")
 	}
 
-	subjFromCert, err := ca.ValidateAndExtractSubject(cert)
+	var verifyOpts []mtls.VerifyCertOption
```

Comment:
nit for more precise allocation:
```suggestion
	verifyOpts := make([]mtls.VerifyCertOption, 0, 1)
```